protocol-units/da/m1/light-node-verifier/src/lib.rs

```diff
@@ -1,44 +1,44 @@
-pub mod v1;
+pub mod celestia;
+pub mod permissioned_signers;
+pub mod signed;
 
 pub use m1_da_light_node_grpc::*;
+use thiserror::Error;
 
-#[tonic::async_trait]
-pub trait Verifier {
-	async fn verify(
-		&self,
-		verification_mode: VerificationMode,
-		blob: &[u8],
-		height: u64,
-	) -> Result<bool, anyhow::Error> {
-		match verification_mode {
-			VerificationMode::Cowboy => self.verify_cowboy(verification_mode, blob, height).await,
-			VerificationMode::ValidatorIn => {
-				self.verifiy_validator_in(verification_mode, blob, height).await
-			}
-			VerificationMode::MOfN => self.verify_m_of_n(verification_mode, blob, height).await,
-		}
+/// Domain error for the transaction pipe task
+#[derive(Debug, Error)]
+pub enum Error {
+	#[error("verifier internal error: {0}")]
+	Internal(String),
+	#[error("verifier validation error: {0}")]
+	Validation(String),
+}
+
+/// thiserror for validation and internal errors
+#[derive(thiserror::Error, Debug)]
+
+/// A verified outcome. Indicates that input of A is verified as valid instance of B, or else invalid instance.
```

Comment:
What's A here?

Reply:
Used to have a Phantom Type `A` indicated `B` was verified from `A`. I removed for immediate convenience. Will updated docs.

Reply:
Actually, that usage is still implied by `VerifierOperations`. I'm not going to revert the type signatures until we have better agreement, but I will just slightly modify the comment.